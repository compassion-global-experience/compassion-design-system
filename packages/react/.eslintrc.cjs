module.exports = {
  extends: [
    'standard-jsx',
    'standard-react',
    'plugin:jsx-a11y/recommended',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
    'plugin:storybook/recommended',
  ],
  env: {
    browser: true,
    es6: true,
    jest: true,
    node: false,
  },
  rules: {
    'jsx-quotes': ['error', 'prefer-double'],
  },
};
