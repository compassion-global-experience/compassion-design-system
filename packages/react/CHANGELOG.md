# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2023-04-08)

### Features

- **react:** css usage and documentation ([11e317c](https://github.com/compassion-global-experience/compassion-design-system/commit/11e317cca4ecdaa385e86d3a5571d7dca554685d))
- **react:** re-export css package ([67549a5](https://github.com/compassion-global-experience/compassion-design-system/commit/67549a58de3de6e02b58c5bdb6f4605001aae783))
- **react:** switch between themes in storybook ([79119ba](https://github.com/compassion-global-experience/compassion-design-system/commit/79119ba48ed3410eaebe9a91ccd076b05c2580c1))
