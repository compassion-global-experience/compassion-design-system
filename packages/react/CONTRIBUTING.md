# Contributing to the `react-web` Package

The `react` package is structured with a component-per-directory convention
inside the `src` directory. Each component directory generally contains 3 items:

- A **TSX file** that describes the component‚Äôs structure and behavior
- A **.stories.ts file** that illustrates most of the ways the component can
  be configured
- An **index.ts** file that exports the TSX component‚Äôs contents

## Workflow

The package uses [Storybook](https://storybook.js.org/docs/react/get-started/introduction)
to facilitate component development and documentation. Storybook uses the files
mentioned above to render components in their various states, and serves as
documentation of those components by surfacing code comments, `propTypes`, and
`defaultProps` values.

To launch Storybook, make sure you‚Äôre in the [/packages/react](/packages/react)
directory and run the command `npm run storybook`. After the project is built,
Storybook will launch in your default browser. New components or modifications
will be displayed in real time! üéâ

The configurations outlined in each component‚Äôs .stories.ts file can be seen in
Storybook‚Äôs **Docs** tab.

## Commit Messages & Versioning

Every package in the compassion-gds monorepo adheres to
[semantic versioning](https://semver.org/). In order to simplify the versioning
and release process, please be sure your commit messages adhere to the
[‚ÄúConventional Commits‚Äù pattern](https://www.conventionalcommits.org/en/v1.0.0/#summary).
You may wish to install the [commitizen/cz-cli](https://github.com/commitizen/cz-cli)
to make this near-bulletproof.

## Making a Pull Request

Once your changes are ready, open a pull request against the `development`
branch of the project, and it will be merged in as quickly as possible!
