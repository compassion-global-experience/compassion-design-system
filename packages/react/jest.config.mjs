/*
 * For a detailed explanation regarding each configuration property, visit:
 * https://jestjs.io/docs/configuration
 */

export default {
  // A preset that is used as a base for Jest's configuration
  preset: 'ts-jest/presets/default-esm',

  // Automatically clear mock calls, instances, contexts and results before every test
  clearMocks: true,

  // Indicates whether the coverage information should be collected while executing the test
  collectCoverage: true,

  // The directory where Jest should output its coverage files
  coverageDirectory: "coverage",

  // Indicates which provider should be used to instrument code for coverage
  coverageProvider: "v8",

  // The test environment that will be used for testing
  testEnvironment: "jsdom",

  // A map from regular expressions to module names or to arrays of module names that allow to stub out resources with a single module
  moduleNameMapper: {
    ".(css|less|scss)$": "identity-obj-proxy",
  },

  // Whether to use watchman for file crawling
  watchman: true,
};
