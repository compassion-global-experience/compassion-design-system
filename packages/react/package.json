{
  "name": "@compassion-gds/react",
  "version": "0.2.0-alpha.2",
  "description": "",
  "type": "module",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "files": [
    "dist"
  ],
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "browser": {
        "default": "./dist/esm/index.js"
      },
      "import": "./dist/esm/index.js",
      "require": "./dist/cjs/index.js"
    },
    "./css/base.css": "./dist/css/reset.css",
    "./css/light.css": "./dist/css/cds-light.css",
    "./css/dark.css": "./dist/css/cds-dark.css",
    "./css/*": "./dist/css/*"
  },
  "scripts": {
    "dev": "rollup -c -w",
    "copy-css": "bash ./scripts/copy-css.sh",
    "prepare": "rollup -c && npm run copy-css",
    "test": "jest",
    "storybook": "storybook dev -p 6007",
    "build:storybook": "storybook build --quiet",
    "percy": "cross-env PERCY_TOKEN=$PERCY_TOKEN_REACT percy storybook ./storybook-static",
    "percy-snapshots": "npm run build:storybook && npm run percy"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/compassion-global-experience/compassion-design-system.git",
    "directory": "packages/react"
  },
  "keywords": [
    "compassion",
    "compassion-gds",
    "gds",
    "react"
  ],
  "author": "Compassion XD",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/compassion-global-experience/compassion-design-system/issues"
  },
  "homepage": "https://github.com/compassion-global-experience/compassion-design-system#readme",
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@compassion-gds/core": "^0.1.0-alpha.0",
    "@rollup/plugin-commonjs": "^24.0.1",
    "@rollup/plugin-node-resolve": "^15.0.1",
    "@rollup/plugin-replace": "^5.0.2",
    "@rollup/plugin-typescript": "^11.1.0",
    "@storybook/addon-actions": "^7.0.2",
    "@storybook/addon-docs": "^7.0.2",
    "@storybook/addon-essentials": "^7.0.2",
    "@storybook/addon-interactions": "^7.0.2",
    "@storybook/addon-links": "^7.0.2",
    "@storybook/preset-react-webpack": "7.0.2",
    "@storybook/react": "7.0.2",
    "@storybook/react-webpack5": "7.0.2",
    "@storybook/testing-library": "^0.0.14-next.1",
    "@testing-library/react": "^13.4.0",
    "@types/jest": "^29.4.0",
    "@types/react": "^18.0.27",
    "babel-loader": "^8.3.0",
    "eslint": "^8.33.0",
    "eslint-config-standard-jsx": "^11.0.0",
    "eslint-config-standard-react": "^13.0.0",
    "eslint-plugin-jsx-a11y": "^6.7.1",
    "eslint-plugin-react": "^7.32.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-storybook": "^0.6.10",
    "jest": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1",
    "phosphor-react": "^1.4.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^3.20.2",
    "rollup-plugin-dts": "^5.1.1",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.2",
    "storybook": "7.0.2",
    "ts-jest": "^29.0.5",
    "webpack": "^5.75.0"
  },
  "peerDependencies": {
    "react": "16.8.x || 17.x || 18.x",
    "react-dom": "16.8.x || 17.x || 18.x"
  }
}
