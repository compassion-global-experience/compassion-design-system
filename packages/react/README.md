# GDS React Components

The Compassion Design System includes a streamlined set of React components for
fast, accessible development of apps and static sites.

## Installing and usage

To install the GDS React package use

```shell
npm install @compassion-gds/react
```

Or `yarn add @compassion-gds/react` if you're using yarn

Import the following css files from the entry point of your application

#### index.js
```js
// Replace {theme} with any of the available themes (e.g. light, dark)
import '@compassion-gds/react/css/base.css';
import '@compassion-gds/react/css/vars/{theme}.css';
```

GDS React components would follow the imported theme.

Sample component usage:

```jsx
import React from 'react';
import { Button } from '@compassion-gds/react';

export const PrimaryButton = (props) => (
  <Button primary size="small" {...props} />
);
```

## Themes

Available options:
- light: `@compassion-gds/react/css/light.css`
- dark: `@compassion-gds/react/css/dark.css`

## Contributing to the `react` Package

The `react` package is structured with a component-per-directory convention
inside the `src` directory. Each component directory generally contains 3 items:

- A **TSX file** that describes the component’s structure and behavior
- A **.stories.ts file** that illustrates most of the ways the component can
  be configured
- An **index.ts** file that exports the TSX component’s contents
