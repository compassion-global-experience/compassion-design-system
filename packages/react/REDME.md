# GDS React Components

The Compassion Design System includes a streamlined set of React components for
fast, accessible development of apps and static sites.

## Installing and usage

To install the GDS React package use

```shell
npm install @compassion-gds/react
```

Or `yarn add @compassion-gds/react` if you're using yarn  
(`@compassion-gds/css` is automatically installed as peer dependency)

Import the following css files from the entry point of your application

#### index.js
```js
// Replace {theme} with any of the available themes (e.g. light, dark)
import '@compassion-gds/css/reset.css';
import '@compassion-gds/css/vars/{theme}.css';
```

GDS React components would follow the imported theme.

Sample component usage:

```jsx
import React from 'react';
import { Button } from '@compassion-gds/react';

export const PrimaryButton = (props) => (
  <Button primary size="small" {...props} />
);
```

## Themes

Available options:
- light: `@compassion-gds/css/vars/light.css`
- dark: `@compassion-gds/css/vars/dark.css`
