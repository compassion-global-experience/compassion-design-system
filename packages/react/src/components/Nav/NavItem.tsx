'use client';

import styles from '@compassion-design-system/core/src/components/Nav/navItem.module.css';
import { getClasses } from '../../utils/classes';
import { cdsIconMap } from '../Icon/Icon';
import { useState, useEffect, useRef, MouseEvent, forwardRef } from 'react';

export interface NavItemProps {
  text: string;
  href: string;
  icon?: keyof typeof cdsIconMap;
  iconDescription?: string;
  target?: '_blank' | '_parent' | '_self' | '_top';
  isDisabled?: boolean;
  isActive?: boolean;
  isContained?: boolean;
  subItems?: NavItemProps[];
  isSubNav?: boolean;
  openSubNav?: boolean;
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
  disableAutoCloseOnClick?: boolean;
}

const NavItem = forwardRef<HTMLAnchorElement, NavItemProps>((props, ref) => {
  const {
    text,
    href,
    icon,
    iconDescription,
    target = '_self',
    isDisabled = false,
    isActive = false,
    isContained = false,
    subItems = [],
    isSubNav = false,
    openSubNav = false,
    onClick,
    disableAutoCloseOnClick = false,
  } = props;

  const [isSubNavOpen, setIsSubNavOpen] = useState<boolean>(openSubNav);
  const subNavRef = useRef(null);

  const subNavItems = subItems.map((item) => {
    return <NavItem {...item} key={item.text} isSubNav />;
  });

  useEffect(() => {
    if (subNavRef.current) {
      if (!isSubNavOpen) {
        subNavRef.current.style.maxHeight = null;
      } else {
        subNavRef.current.style.maxHeight =
          subNavRef.current.scrollHeight + 'px';
      }
    }
  }, [isSubNavOpen]);

  useEffect(() => {
    setIsSubNavOpen(openSubNav);
  }, [openSubNav]);

  const handleOnClick = (e) => {
    if (onClick) {
      onClick(e);
    }
    if (!disableAutoCloseOnClick) {
      setIsSubNavOpen(!isSubNavOpen);
    }
  };

  const navItemContainerClasses = ['cds-navItemContainer'];
  if (isContained) {
    navItemContainerClasses.push('cds-navItemContainer--contained');
  }

  const navItemClasses = ['cds-navItem'];
  if (isDisabled) {
    navItemClasses.push('cds-navItem--disabled');
  } else {
    if (isActive) {
      navItemClasses.push('cds-navItem--active');
    }
  }

  const NavIcon = cdsIconMap[icon];
  const ExpandSubsectionIcon = cdsIconMap.caretDown;

  return (
    <div className={getClasses(styles, navItemContainerClasses)}>
      <div className={getClasses(styles, navItemClasses)}>
        {icon && NavIcon && (
          <NavIcon
            className={getClasses(styles, ['cds-navItemSvg'])}
            size={isSubNav ? 16 : 20}
            alt={iconDescription}
          />
        )}
        <div className={getClasses(styles, ['cds-navItemText'])}>
          {isDisabled && (
            <span className={getClasses(styles, ['cds-navItemText__disabled'])}>
              {text}
            </span>
          )}{' '}
          {!isDisabled && (
            <a
              ref={ref}
              href={href}
              target={target}
              aria-disabled="false"
              onClick={handleOnClick}
            >
              {text}
            </a>
          )}
        </div>
        {!isSubNav && subItems.length > 0 && (
          <button
            aria-label={
              isSubNavOpen ? 'Hide sub navigation' : 'Show sub navigation'
            }
            className={getClasses(styles, [
              'cds-navItemExpander',
              isSubNavOpen ? 'cds-navItemExpander--opened' : '',
            ])}
            onClick={() => {
              setIsSubNavOpen(!isSubNavOpen);
            }}
          >
            <ExpandSubsectionIcon
              className={getClasses(styles, ['cds-navItemSvg'])}
              size={isSubNav ? 16 : 20}
            />
          </button>
        )}
      </div>
      {!isSubNav && subItems.length > 0 && (
        <div
          ref={subNavRef}
          className={getClasses(styles, ['cds-subNavWrapper'])}
        >
          <div className={getClasses(styles, ['cds-subNavContainer'])}>
            {subNavItems}
          </div>
        </div>
      )}
    </div>
  );
});

NavItem.displayName = 'NavItem';
export default NavItem;
