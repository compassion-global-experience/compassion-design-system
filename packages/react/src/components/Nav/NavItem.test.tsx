import { render } from '@testing-library/react';
import NavItem, { NavItemProps } from './NavItem';

const testProps: NavItemProps = {
  text: 'Hello World',
  href: '#',
  icon: 'house',
  iconDescription: 'Website home',
};

const testSubItems: NavItemProps[] = [
  { text: 'Sub Nav Item 1', href: '#' },
  { text: 'Sub Nav Item 2', href: '#' },
  { text: 'Sub Nav Item 3', href: '#' },
];

describe('NavItem', () => {
  it('should render a single nav item with no subnav button or items', () => {
    const { container } = render(<NavItem {...testProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should render a nav item with subnav closed by default', () => {
    const { container } = render(
      <NavItem {...testProps} subItems={testSubItems} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render a nav item with subnav open', () => {
    const { container } = render(
      <NavItem {...testProps} subItems={testSubItems} openSubNav />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render a contained navitem', () => {
    const { container } = render(
      <NavItem {...testProps} subItems={testSubItems} isContained />,
    );
    expect(container).toMatchSnapshot();
  });
});
