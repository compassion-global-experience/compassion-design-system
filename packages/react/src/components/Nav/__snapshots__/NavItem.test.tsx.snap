// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem should render a contained navitem 1`] = `
<div>
  <div
    class="cds-navItemContainer cds-navItemContainer--contained"
  >
    <div
      class="cds-navItem"
    >
      <div
        class="cds-navItem-icon"
      >
        <svg
          class="cds-navItemSvg"
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Website home
          </title>
          <path
            d="M240,208H224V115.55a16,16,0,0,0-5.17-11.78l-80-75.48a1.14,1.14,0,0,1-.11-.11,16,16,0,0,0-21.53,0l-.11.11L37.17,103.77A16,16,0,0,0,32,115.55V208H16a8,8,0,0,0,0,16H240a8,8,0,0,0,0-16ZM48,115.55l.11-.1L128,40l79.9,75.43.11.1V208H160V160a16,16,0,0,0-16-16H112a16,16,0,0,0-16,16v48H48ZM144,208H112V160h32Z"
          />
        </svg>
      </div>
      <div
        class="cds-navItemText"
      >
         
        <a
          aria-disabled="false"
          href="#"
          target="_self"
        >
          Hello World
        </a>
      </div>
      <button
        aria-label="Show sub navigation"
        class="cds-navItemExpander"
      >
        <svg
          class="cds-navItemSvg"
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
          />
        </svg>
      </button>
    </div>
    <div
      class="cds-subNavWrapper"
    >
      <div
        class="cds-subNavContainer"
      >
        <div
          class="cds-navItemContainer"
        >
          <div
            class="cds-navItem"
          >
            <div
              class="cds-navItemText"
            >
               
              <a
                aria-disabled="false"
                href="#"
                target="_self"
              >
                Sub Nav Item 1
              </a>
            </div>
          </div>
        </div>
        <div
          class="cds-navItemContainer"
        >
          <div
            class="cds-navItem"
          >
            <div
              class="cds-navItemText"
            >
               
              <a
                aria-disabled="false"
                href="#"
                target="_self"
              >
                Sub Nav Item 2
              </a>
            </div>
          </div>
        </div>
        <div
          class="cds-navItemContainer"
        >
          <div
            class="cds-navItem"
          >
            <div
              class="cds-navItemText"
            >
               
              <a
                aria-disabled="false"
                href="#"
                target="_self"
              >
                Sub Nav Item 3
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavItem should render a nav item with subnav closed by default 1`] = `
<div>
  <div
    class="cds-navItemContainer"
  >
    <div
      class="cds-navItem"
    >
      <div
        class="cds-navItem-icon"
      >
        <svg
          class="cds-navItemSvg"
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Website home
          </title>
          <path
            d="M240,208H224V115.55a16,16,0,0,0-5.17-11.78l-80-75.48a1.14,1.14,0,0,1-.11-.11,16,16,0,0,0-21.53,0l-.11.11L37.17,103.77A16,16,0,0,0,32,115.55V208H16a8,8,0,0,0,0,16H240a8,8,0,0,0,0-16ZM48,115.55l.11-.1L128,40l79.9,75.43.11.1V208H160V160a16,16,0,0,0-16-16H112a16,16,0,0,0-16,16v48H48ZM144,208H112V160h32Z"
          />
        </svg>
      </div>
      <div
        class="cds-navItemText"
      >
         
        <a
          aria-disabled="false"
          href="#"
          target="_self"
        >
          Hello World
        </a>
      </div>
      <button
        aria-label="Show sub navigation"
        class="cds-navItemExpander"
      >
        <svg
          class="cds-navItemSvg"
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
          />
        </svg>
      </button>
    </div>
    <div
      class="cds-subNavWrapper"
    >
      <div
        class="cds-subNavContainer"
      >
        <div
          class="cds-navItemContainer"
        >
          <div
            class="cds-navItem"
          >
            <div
              class="cds-navItemText"
            >
               
              <a
                aria-disabled="false"
                href="#"
                target="_self"
              >
                Sub Nav Item 1
              </a>
            </div>
          </div>
        </div>
        <div
          class="cds-navItemContainer"
        >
          <div
            class="cds-navItem"
          >
            <div
              class="cds-navItemText"
            >
               
              <a
                aria-disabled="false"
                href="#"
                target="_self"
              >
                Sub Nav Item 2
              </a>
            </div>
          </div>
        </div>
        <div
          class="cds-navItemContainer"
        >
          <div
            class="cds-navItem"
          >
            <div
              class="cds-navItemText"
            >
               
              <a
                aria-disabled="false"
                href="#"
                target="_self"
              >
                Sub Nav Item 3
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavItem should render a nav item with subnav open 1`] = `
<div>
  <div
    class="cds-navItemContainer"
  >
    <div
      class="cds-navItem"
    >
      <div
        class="cds-navItem-icon"
      >
        <svg
          class="cds-navItemSvg"
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Website home
          </title>
          <path
            d="M240,208H224V115.55a16,16,0,0,0-5.17-11.78l-80-75.48a1.14,1.14,0,0,1-.11-.11,16,16,0,0,0-21.53,0l-.11.11L37.17,103.77A16,16,0,0,0,32,115.55V208H16a8,8,0,0,0,0,16H240a8,8,0,0,0,0-16ZM48,115.55l.11-.1L128,40l79.9,75.43.11.1V208H160V160a16,16,0,0,0-16-16H112a16,16,0,0,0-16,16v48H48ZM144,208H112V160h32Z"
          />
        </svg>
      </div>
      <div
        class="cds-navItemText"
      >
         
        <a
          aria-disabled="false"
          href="#"
          target="_self"
        >
          Hello World
        </a>
      </div>
      <button
        aria-label="Hide sub navigation"
        class="cds-navItemExpander cds-navItemExpander--opened"
      >
        <svg
          class="cds-navItemSvg"
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
          />
        </svg>
      </button>
    </div>
    <div
      class="cds-subNavWrapper"
      style="max-height: 0px;"
    >
      <div
        class="cds-subNavContainer"
      >
        <div
          class="cds-navItemContainer"
        >
          <div
            class="cds-navItem"
          >
            <div
              class="cds-navItemText"
            >
               
              <a
                aria-disabled="false"
                href="#"
                target="_self"
              >
                Sub Nav Item 1
              </a>
            </div>
          </div>
        </div>
        <div
          class="cds-navItemContainer"
        >
          <div
            class="cds-navItem"
          >
            <div
              class="cds-navItemText"
            >
               
              <a
                aria-disabled="false"
                href="#"
                target="_self"
              >
                Sub Nav Item 2
              </a>
            </div>
          </div>
        </div>
        <div
          class="cds-navItemContainer"
        >
          <div
            class="cds-navItem"
          >
            <div
              class="cds-navItemText"
            >
               
              <a
                aria-disabled="false"
                href="#"
                target="_self"
              >
                Sub Nav Item 3
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavItem should render a single nav item with no subnav button or items 1`] = `
<div>
  <div
    class="cds-navItemContainer"
  >
    <div
      class="cds-navItem"
    >
      <div
        class="cds-navItem-icon"
      >
        <svg
          class="cds-navItemSvg"
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Website home
          </title>
          <path
            d="M240,208H224V115.55a16,16,0,0,0-5.17-11.78l-80-75.48a1.14,1.14,0,0,1-.11-.11,16,16,0,0,0-21.53,0l-.11.11L37.17,103.77A16,16,0,0,0,32,115.55V208H16a8,8,0,0,0,0,16H240a8,8,0,0,0,0-16ZM48,115.55l.11-.1L128,40l79.9,75.43.11.1V208H160V160a16,16,0,0,0-16-16H112a16,16,0,0,0-16,16v48H48ZM144,208H112V160h32Z"
          />
        </svg>
      </div>
      <div
        class="cds-navItemText"
      >
         
        <a
          aria-disabled="false"
          href="#"
          target="_self"
        >
          Hello World
        </a>
      </div>
    </div>
  </div>
</div>
`;
